spec/forms/totp_verification_form_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe TotpVerificationForm do
+  describe '#submit' do
+    context 'when the form is valid' do
+      it 'returns FormResponse with success: true' do
+        user = build_stubbed(:user)
+        code = '123456'
+        form = TotpVerificationForm.new(user, code)
+        result = instance_double(FormResponse)
+
+        allow(user).to receive(:authenticate_totp).and_return(true)
+
+        expect(FormResponse).to receive(:new).
+          with(success: true, errors: {}, extra: { multi_factor_auth_method: 'totp' }).
+          and_return(result)
+        expect(form.submit).to eq result
+      end
+    end
+
+    context 'when the form is invalid' do
+      it 'returns false' do
+        user = build_stubbed(:user)
+        code = '123456'
+        form = TotpVerificationForm.new(user, code)
+        result = instance_double(FormResponse)
+
+        allow(user).to receive(:authenticate_totp).and_return(false)
+
+        expect(FormResponse).to receive(:new).
+          with(success: false, errors: {}, extra: { multi_factor_auth_method: 'totp' }).
+          and_return(result)
+        expect(form.submit).to eq result
```

Comment:
the spec description says it returns false, but it looks like here we are expecting an instance double - should the spec description be changed, or expectation changed? same for spec above